internal/http/handler.go

```diff
@@ -120,6 +121,41 @@ func HandleVersion(version string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) { fmt.Fprintf(w, `{"version":"%s"}`, version) }
 }
 
+// HandleStatus returns a handler function that returns status
+// information, like server version and server up-time, as JSON
+// object to the client.
+//
+// The handler function should not return any internal data - like
+// keystore backend, cache expiry, etc.
+func HandleStatus(version string, certificate *Certificate) http.HandlerFunc {
+	type Status struct {
+		Version string        `json:"version"`
+		UpTime  time.Duration `json:"uptime"`
+
+		Certificate struct {
+			Expiry   time.Duration `json:"expiry"`
+			DNSNames []string      `json:"dns,omitempty"`
+			IPs      []net.IP      `json:"ip,omitempty"`
+		} `json:"certificate"`
+	}
+	var startTime = time.Now()
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+
+		var status = Status{
+			Version: version,
+			UpTime:  time.Since(startTime).Truncate(time.Second),
+		}
+		if cert, err := certificate.GetCertificate(nil); cert != nil && err == nil {
+			status.Certificate.Expiry = time.Until(cert.Leaf.NotAfter).Truncate(time.Second)
+			status.Certificate.DNSNames = cert.Leaf.DNSNames
+			status.Certificate.IPs = cert.Leaf.IPAddresses
```

Comment:
On second thought should we log an error here? or send it to client?

Reply:
This should not happen since a) KES requires TLS and b) `GetCertificate` should always return a certificate.
However, we don't provide a `tls.ClientHello` (but `nil` since we don't have one) and `GetCertificate` may do things like SNI certificate mapping.  
If `certificate.GetCertificate(nil)` returns `nil` / an error then `GetCertificate` sporadically fails to return a certificate. It has to succeed at least sometimes - otherwise, the client could not even connect to the server. 

However, we could log an error message. Sending an error to the client may be odd since from client perspective it will look tike:
 1. Client to Server connection succeeded. Therefore `GetCertificate` returned a certificate
 2. Then `GetCertificate` failed and the server returned an error.
 So, for the client it would look like the server first sends its TLS certificate and then the server claims to not have a TLS certificate.

Reply:
Added error logging for this case.